### PR TITLE
test: Code Coverage: Add Edit Mileage: #8 Part 3

### PR DESCRIPTION
**Pull request by jayfyle**
### <samp>🤖 Generated by Copilot at ca61c8d</samp>

This pull request adds and updates unit tests and mock data for the mileage and report features of the app. It also refactors the code for displaying the mileage rates options in the `add-edit-mileage` page, and removes some unused code.

### <samp>🤖 Generated by Copilot at ca61c8d</samp>

> _Sing, O Muse, of the agile developers who toiled_
> _To craft the app for tracking expenses and mileages,_
> _And how they tested their code with mock data and spies,_
> _And how they refactored the `add-edit-mileage` page with skill._

### <samp>🤖 Generated by Copilot at ca61c8d</samp>

*  Add and refactor methods for getting mileage rates options ([link](https://github.com/fylein/fyle-mobile-app/pull/2339/files?diff=unified&w=0#diff-04c216da6200af711aa0f53bcd7e253086762b5d123b0729fac3b1862ec5fc3eR1261-R1277), [link](https://github.com/fylein/fyle-mobile-app/pull/2339/files?diff=unified&w=0#diff-04c216da6200af711aa0f53bcd7e253086762b5d123b0729fac3b1862ec5fc3eL1371-R1388), [link](https://github.com/fylein/fyle-mobile-app/pull/2339/files?diff=unified&w=0#diff-04c216da6200af711aa0f53bcd7e253086762b5d123b0729fac3b1862ec5fc3eL1390-L1391))
* Add and invoke test cases for the mileage feature of the app ([link](https://github.com/fylein/fyle-mobile-app/pull/2339/files?diff=unified&w=0#diff-df40b6e711e89d5e80e6f3867394ee96bbff1eeae11eef42db09ce0b3f1821fbR63), [link](https://github.com/fylein/fyle-mobile-app/pull/2339/files?diff=unified&w=0#diff-df40b6e711e89d5e80e6f3867394ee96bbff1eeae11eef42db09ce0b3f1821fbR77), [link](https://github.com/fylein/fyle-mobile-app/pull/2339/files?diff=unified&w=0#diff-df40b6e711e89d5e80e6f3867394ee96bbff1eeae11eef42db09ce0b3f1821fbR220), [link](https://github.com/fylein/fyle-mobile-app/pull/2339/files?diff=unified&w=0#diff-df40b6e711e89d5e80e6f3867394ee96bbff1eeae11eef42db09ce0b3f1821fbR447))
* Add mock data for recently used mileages and report options ([link](https://github.com/fylein/fyle-mobile-app/pull/2339/files?diff=unified&w=0#diff-6512bbe410b564ad644cd215bf5f483358a11ae57e943d817a0da542b98d8cf3R190-R207), [link](https://github.com/fylein/fyle-mobile-app/pull/2339/files?diff=unified&w=0#diff-178ba4cc3ac9d9b1a14067c05785ec609622a12ac6cd6eed96bbc0b5dd2a221bR274-R520))

## Clickup
https://app.clickup.com/t/85ztjmtfw

## Code Coverage
`94.37% Statements 788/835`
`84.92% Branches 603/710`
`95.54% Functions 279/292`
`94.22% Lines 750/796`

## UI Preview
Please add screenshots for UI changes

### CODE REVIEW

File: src/app/fyle/add-edit-mileage/add-edit-mileage-5.spec.ts [hunk 1]
```diff
@@ -0,0 +1,413 @@
...
+      component.selectedCostCenter$ = new BehaviorSubject(null);
+      fixture.detectChanges();
+    });
+
+    it('should create', () => {
+      expect(component).toBeTruthy();
+    });
+
+    describe('ionViewWillEnter():', () => {
+      it('should setup class variables', fakeAsync(() => {
```
Comment (Dimple16) [inline src/app/fyle/add-edit-mileage/add-edit-mileage-5.spec.ts:215]: check if we can break this test case into smaller parts? too long now

**Review by Dimple16 — CHANGES_REQUESTED** [review]
added comments

File: src/app/fyle/add-edit-mileage/add-edit-mileage.page.ts [hunk 2]
```diff
@@ -1368,27 +1385,12 @@ export class AddEditMileagePage implements OnInit {
...
-        })
-      )
-    );
+    this.getMileageRatesOptions();
 
     this.etxn$ = iif(() => this.mode === 'add', this.getNewExpense(), this.getEditExpense());
 
     this.setupTfcDefaultValues();
 
-    this.isAmountDisabled$ = this.etxn$.pipe(map((etxn) => !!etxn.tx.admin_amount));
```
[the remark this answers is not in the document]
Comment (jayfyle) [reply src/app/fyle/add-edit-mileage/add-edit-mileage.page.ts:1390]: This is used, but this particular line was repeated, hence removed it.

File: src/app/fyle/add-edit-mileage/add-edit-mileage-5.spec.ts (hunk 1 above)
Comment (jayfyle) [reply src/app/fyle/add-edit-mileage/add-edit-mileage-5.spec.ts:215]: Will fix this in the next PR, with both the test cases, will make it easier to group common mocks.

**Review by OmkarJ13 — COMMENTED** [review]
There are few merge conflicts, can you pls re-assign once the conflicts are fixed, will review afterwards

**Review by Dimple16 — APPROVED** [review]

**Review by OmkarJ13 — APPROVED** [review]

**Review by Chethan-Fyle — APPROVED** [review]

**Pull request merged**